build/xtask/src/dist.rs

```diff
@@ -549,6 +555,53 @@ pub fn package(
     Ok(allocated)
 }
 
+// generate file with hash of expected flash contents
+fn write_fwid(
+    cfg: &PackageConfig,
+    image_name: &String,
+    flash: &Range<u32>,
+    archive_name: &PathBuf,
+) -> Result<()> {
+    let mut archive = hubtools::RawHubrisArchive::load(archive_name)
+        .context("loading archive with hubtools")?;
+
+    let bin = archive
+        .extract_file("img/final.bin")
+        .context("extracting final.bin after signing & caboosing")?;
+
+    let chip_name = Path::new(&cfg.toml.chip);
+
+    // determine FWID calculation method from chip (directory) name
+    let pad = match chip_name.file_name().and_then(OsStr::to_str) {
+        Some("stm32h7") => {
+            // all unprogrammed flash is read as 0xff
+            Some(vec![
+                0xff_u8;
+                flash.end as usize - flash.start as usize - bin.len()
+            ])
+        }
+        Some(_) => None,
+        None => bail!("Failed to get file name of {}", chip_name.display()),
+    };
+
+    let mut sha = Sha3_256::new();
+    sha.update(&bin);
+
+    if let Some(pad) = pad {
+        sha.update(&pad);
+    }
+
+    let digest = sha.finalize();
+
+    let mut file = File::create(&cfg.img_file("final.fwid", image_name))?;
+    writeln!(file, "{}", hex::encode(&digest))?;
+
+    archive.add_file("img/final.fwid", digest.as_bytes())?;
```

Comment:
I'm not quite sure why we're creating the file twice, is the hubris one not sufficient? Aren't the two files different since one is hex encoded and the other is the raw bytes?

Reply:
Ah that's a good catch: I had intended for the contents of these two files to be identical. I'll fix that up. As to why I create a file on disk and one in the archive: the `final.bin` is manipulated / caboosed in the archive after creation.

Reply:
Just pushed a change to make the `final.fwid` files on disk and in the archive identical: both now contain the hex encoded digest.